JavadocStyleGuide.md

```diff
@@ -0,0 +1,223 @@
+# Gradle Javadoc Style Guide
+
+## 1.1 Formatting
+
+### 1.1.1 General form
+
+The basic formatting of Javadoc blocks is as seen in this example:
+
+```java
+/**
+ * Returns an Image object that can then be painted on the screen.
+ * The url argument must specify an absolute <a href="#{@link}">{@link URL}</a>. The name
+ * argument is a specifier that is relative to the url argument.
+ * <p>
+ * This method always returns immediately, whether or not the
+ * image exists. When this applet attempts to draw the image on
+ * the screen, the data will be loaded. The graphics primitives
+ * that draw the image will incrementally paint on the screen.
+ *
+ * @param  url  an absolute URL giving the base location of the image
+ * @param  name the location of the image, relative to the url argument
+ * @return      the image at the specified URL
+ * @see         Image
```

Comment:
All of these examples have the description text left-aligned. This can be very tedious when adding new parameters or changing the names of existing parameters. 

Consider the use-case of changing `name` to `imageName` (for whatever reason). Even though I just wanted to modify a single line, I would also need to modify three other lines. 

There are a few downsides to this:
1. Increases complexity of code review. Only `name` -> `imageName` changed, but three other unrelated lines show up in the code diff. 
2. Reduces accuracy of git history. Git history is important for determining when features got added, when bugs were introduced, or determining context of existing code. Causing unrelated code changes like this (renaming `name` affecting surrounding lines) can make it more difficult to track changes in git history, as initially it appears that `@param url` and `@return` changed when in reality their whitespace was modified due to the renaming of `name`. 

I have not seen any official notes in this page saying we _should_ have these text descriptions left-aligned, but all examples show this as the case. I do not think we should enforce alignment of these text blocks.

Reply:
Please let's not do this.  Single spaces between `@param`, the name, and the desc.  Likewise for the other tags.

Reply:
<p>, <br> and and empty * line are all line break in IntelliJ IDEA. Style guide has been updated.

Reply:
Style guide has been updated to continue with the curent convention (no left alignment).